pkg/envinfo/url.go

```diff
@@ -15,31 +14,45 @@ import (
 	"github.com/pkg/errors"
 )
 
-// GetPorts returns the ports, returns default if nil
-func (ei *EnvInfo) GetPorts() ([]string, error) {
+//getPorts gets the ports from devfile
+func (ei *EnvInfo) getPorts(container string) ([]string, error) {
 	var portList []string
-
+	portMap := make(map[string]bool)
 	containerComponents, err := ei.devfileObj.Data.GetDevfileContainerComponents(common.DevfileOptions{})
 	if err != nil {
-		return portList, err
+		return nil, err
 	}
-
-	portMap := make(map[string]bool)
-
+	containerExists := false
 	for _, component := range containerComponents {
-		for _, endpoint := range component.Container.Endpoints {
-			portMap[strconv.FormatInt(int64(endpoint.TargetPort), 10)] = true
+		if container == "" || container == component.Name {
+			containerExists = true
+			for _, endpoint := range component.Container.Endpoints {
+				portMap[strconv.FormatInt(int64(endpoint.TargetPort), 10)] = true
+			}
 		}
 	}
-
+	if !containerExists {
+		return portList, fmt.Errorf("container named %s does not exist", container)
+	}
 	for port := range portMap {
 		portList = append(portList, port)
 	}
-
-	sort.Strings(portList)
```

Comment:
Why not sorting the ports anymore? That could break some tests and users scenarios

Reply:
Yea i guess this can make tests flaky, adding it back